src/opm/input/eclipse/Parser/Parser.cpp

```diff
@@ -1582,13 +1585,40 @@ std::vector<std::string> Parser::getAllDeckNames () const {
                     // ignore unknown keywords for now (i.e. they can appear in any section)
                     continue;
 
-                const auto& parserKeyword = parser.getParserKeywordFromDeckName( curKeywordName );
-                if (ensureKeywordSectionAffiliation && !parserKeyword.isValidSection(curSectionName)) {
-                    std::string msg =
-                        "The keyword '"+curKeywordName+"' is located in the '"+curSectionName
-                        +"' section where it is invalid";
-                    OpmLog::warning(Log::fileMessage(curKeyword.location(), msg) );
-                    deckValid = false;
+                const bool isOperateKeyword =
+                    Fieldprops::keywords::is_oper_keyword(curKeywordName);
+
+                auto checker = [&errorGuard, &deckValid, &parser, curSectionName,
+                                ensureKeywordSectionAffiliation, errorKey]
+                    (const std::string& curKeywordName, const KeywordLocation& location)
```

Comment:
I didn't notice this before, but this parameter (`curKeywordName`) shadows [an outer variable](https://github.com/blattms/opm-common/blob/e5c44baf38555cba8f3417b46050330d03025cdc/src/opm/input/eclipse/Parser/Parser.cpp#L1581).  This, in turn, [generates an additional warning](https://ci.opm-project.org/job/opm-common-PR-builder/5993/gcc/new/).  Please fix.

Reply:
good catch. see #3624